modules/core/src/main/scala/com.snowplowanalytics.snowplow/scalatracker/Emitter.scala

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2015-2018 Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+package com.snowplowanalytics.snowplow.scalatracker
+
+import java.util.UUID
+
+/**
+ * Emitters are entities in charge of transforming events sent from tracker
+ * into actual HTTP requests (IO), which includes:
+ * + Async/Multi-threading
+ * + Queuing `EmitterPayload`
+ * + Transforming `EmitterPayload` into Bytes
+ * + Backup queue and callbacks
+ */
+trait Emitter[F[_]] {
+  import Emitter._
+
+  /**
+   * Method called to send an event from the tracker to the emitter
+   *
+   * @param event Fully assembled event
+   */
+  def send(event: EmitterPayload): F[Unit]
+
+  def getNanoTime: F[Long]
```

Comment:
@chuwy I took a bit different approach to what we discussed - instead of separate implicit algebra I just put these methods into `Emitter`, changed type of `emitters` in `Tracker` from `List` to `NonEmptyList` so that we can always call `emitters.head.getNanoTime`. I think it's simpler, what's your view on this?